activation/handler.go

```diff
@@ -548,18 +534,40 @@ func (h *Handler) HandleGossipAtx(ctx context.Context, peer p2p.Peer, msg []byte
 	if errors.Is(err, errKnownAtx) && peer == h.local {
 		return nil
 	}
+
+	// broadcast malfeasance proof last as the verification of the proof will take place
+	// in the same goroutine
+	if proof != nil {
+		gossip := types.MalfeasanceGossip{
+			MalfeasanceProof: *proof,
+		}
+		encodedProof, err := codec.Encode(&gossip)
+		if err != nil {
+			h.log.With().Fatal("failed to encode malfeasance gossip", log.Err(err))
+		}
+		if err = h.publisher.Publish(ctx, pubsub.MalfeasanceProof, encodedProof); err != nil {
+			h.log.With().Error("failed to broadcast malfeasance proof", log.Err(err))
+			return fmt.Errorf("broadcast atx malfeasance proof: %w", err)
```

Comment:
is the log 1 line above necessary if we return same error up the stack?

Reply:
The error message from the handlers was being ignored in pubsub code:
https://github.com/spacemeshos/go-spacemesh/blob/6cd369d2a2cae7e164d9356005325257b569208a/p2p/pubsub/wrapper.go#L41-L51
I've added debug logging there, but IIUC failing to be able to broadcast the malfeasance proof should be more than a silent validation error by default, so I kept logging there, too.